sdks/java/io/debezium/src/main/java/org/apache/beam/io/debezium/KafkaConnectUtils.java

```diff
@@ -74,4 +77,45 @@ public static Schema.FieldType beamSchemaTypeFromKafkaType(
                 "Unable to convert Kafka field schema %s to Beam Schema", kafkaFieldSchema));
     }
   }
+
+  public static SourceRecordMapper<Row> beamRowFromSourceRecordFn(final Schema recordSchema) {
+    return new SourceRecordMapper<Row>() {
+      @Override
+      public Row mapSourceRecord(SourceRecord sourceRecord) throws Exception {
+        return beamRowFromKafkaStruct((Struct) sourceRecord.value(), recordSchema);
+      }
+
+      private Row beamRowFromKafkaStruct(Struct kafkaStruct, Schema beamSchema) {
+        Row.Builder rowBuilder = Row.withSchema(beamSchema);
+        for (Schema.Field f : beamSchema.getFields()) {
+          Object structField = kafkaStruct.getWithoutDefault(f.getName());
+          switch (kafkaStruct.schema().field(f.getName()).schema().type()) {
+            case ARRAY:
+            case MAP:
+              // TODO(pabloem): Handle nested structs
+              throw new IllegalArgumentException("UNABLE TO CONVERT FIELD " + f);
+            case STRUCT:
+              Schema fieldSchema = f.getType().getRowSchema();
+              if (fieldSchema == null) {
+                throw new IllegalArgumentException(
+                    String.format("Improper schema for Beam record: %s has no row schema to build a Row from.", f.getName());
```

Comment:
```suggestion
                    String.format("Improper schema for Beam record: %s has no row schema to build a Row from.", f.getName()));
```
Looks like I missed a parenthesis here

Reply:
added!